src/config/searchWorker.ts

```diff
@@ -16,29 +16,86 @@
 
 /* eslint-disable no-restricted-globals */
 export default () => {
-    function debounceSearch(callback: (...args: any[]) => void) {
-        let timeout
-        return (...args: any[]): void => {
+    /**
+     * Compares two values for sorting.
+     * @param  a - The first value to compare.
+     * @param  b - The second value to compare.
+     * @returns  The comparison result (-1 if a < b, 0 if a = b, 1 if a > b).
+     */
+    const comparator = (a: string | number | object, b: string | number | object) => {
+        let result = 0
+        if (a < b) {
+            result = -1
+        } else if (a > b) {
+            result = 1
+        }
+        return result
+    }
+
+    /**
+     * Dynamically sorts an array of objects based on a specified property and sorting order.
+     * @param  property - The property by which to sort the objects.
+     * @param  sortOrder - The sorting order ('ASC' for ascending, 'DESC' for descending).
+     * @returns A sorting function.
+     */
+    const dynamicSort = (property: string, sortOrder: 'ASC' | 'DESC') => {
+        const sortingOrder = sortOrder === 'ASC' ? 1 : -1
+
+        /**
+         * Sorting function for comparing two objects based on a specified property.
+         * @param a - The first object to compare.
+         * @param b - The second object to compare.
+         * @returns The comparison result (-1 if a < b, 0 if a = b, 1 if a > b).
+         */
+        return (a: Record<string, string | number | object>, b: Record<string, string | number | object>) => {
+            const valueA = a[property]
+            const valueB = b[property]
+
+            if (!Number.isNaN(Number(valueA)) && !Number.isNaN(Number(valueB))) {
+                return comparator(Number(valueA), Number(valueB)) * sortingOrder
+            }
+
+            return comparator(valueA, valueB) * sortingOrder
+        }
+    }
+
+    const debounceSearch = <T extends unknown[]>(callback: (...args: T) => void) => {
+        let timeout: NodeJS.Timeout
+        return (...args: T) => {
             clearTimeout(timeout)
             timeout = setTimeout(() => {
                 callback.apply(self, args)
             }, 300)
         }
     }
 
-    function getFilteredList({ searchText, list }) {
+    const getFilteredList = ({
+        searchText,
+        list,
+        sortBy,
+        sortOrder,
+    }: {
+        searchText: string
+        list: unknown[]
+        sortBy: string
+        sortOrder: 'ASC' | 'DESC'
+    }) => {
         const searchTextLowerCased = searchText.toLowerCase()
-        if (searchTextLowerCased === '' || !list?.length) {
-            self.postMessage(list)
-            return
-        }
-        self.postMessage(
-            list.filter((item) =>
+        let filteredList = [...list]
```

Comment:
We can consider using structured clone instead because of the possibility of nested objects

Reply:
there won't be nested objects only strings, number